### PR TITLE
fix #4944 command not found error error

### LINKED ISSUE #4944: Error in CommandNotFoundError
Commit 809532b7cc5ec1ea18f1aa565025d7c5599185b3 has broken the class CommandNotFoundError. Examples to trigger both code paths calling CommandNotFoundError():

```
(root) [root@d4cb01a1b5b6 ~]# conda activate
Traceback (most recent call last):
  File "/conda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/conda/lib/python2.7/site-packages/conda/cli/main.py", line 161, in main
    raise CommandNotFoundError(argv1, message)
TypeError: __init__() takes exactly 2 arguments (3 given)
```

```
(root) [root@d4cb01a1b5b6 ~]# conda -unknown unknown                                                                                                                                                               
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Current conda install:

               platform : linux-64
          conda version : 4.3.14
       conda is private : False
      conda-env version : 4.3.14
    conda-build version : not installed
         python version : 2.7.13.final.0
       requests version : 2.12.4
       root environment : /conda  (writable)
    default environment : /conda
       envs directories : /conda/envs
                          /root/.conda/envs
          package cache : /conda/pkgs
                          /root/.conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /root/.condarc
           offline mode : False
             user-agent : conda/4.3.14 requests/2.12.4 CPython/2.7.13 Linux/4.8.6-300.fc25.x86_64 CentOS/6.8 glibc/2.12
                UID:GID : 0:0

`$ /conda/bin/conda -unknown unknown`

    Traceback (most recent call last):
      File "/conda/lib/python2.7/site-packages/conda/exceptions.py", line 573, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/conda/lib/python2.7/site-packages/conda/cli/main.py", line 119, in _main
        args = p.parse_args(args)
      File "/conda/lib/python2.7/site-packages/conda/cli/conda_argparse.py", line 173, in parse_args
        return super(ArgumentParser, self).parse_args(*args, **kwargs)
      File "/conda/lib/python2.7/argparse.py", line 1701, in parse_args
        args, argv = self.parse_known_args(args, namespace)
      File "/conda/lib/python2.7/argparse.py", line 1740, in parse_known_args
        self.error(str(err))
      File "/conda/lib/python2.7/site-packages/conda/cli/conda_argparse.py", line 146, in error
        raise CommandNotFoundError(cmd, message)
    TypeError: __init__() takes exactly 2 arguments (3 given)
```

### PR DESCRIPTION
**Pull request by kalefranz**
resolves #4944

### CODE REVIEW

**Comment by kalefranz** [comment]
@goanpeca What do I do to label this PR? Or do I need to do anything?

**Comment by goanpeca** [comment]
@kalefranz for PRs all we need is to add the correct milestone

File: conda/exceptions.py [hunk 1]
```diff
@@ -159,7 +159,32 @@ def __init__(self, target_path, incompatible_package_dists, context):
 
 class CommandNotFoundError(CondaError):
     def __init__(self, command):
-        message = "Conda could not find the command: '%(command)s'"
+        build_commands = {
+            'build',
```
Comment (goanpeca) [inline conda/exceptions.py]: @kalefranz why not reuse the build commands dict here https://github.com/conda/conda/pull/4963/files#diff-aa5c1e6732e4802903e91fe909341fd0R19

Comment (goanpeca) [inline conda/exceptions.py]: Or maybe you forgot to erase that one?

Comment (kalefranz) [inline conda/exceptions.py]: yeah meant to erase that one I think

Comment (kalefranz) [inline conda/exceptions.py]: erased in one of the commits below

**Pull request merged**